### PR TITLE
Explicit message error for table that doesn't exist

### PR DESCRIPTION
**Pull request by carlosfrodrigues**
The solution I gave was change the variable ErrTableNotFound to a  function with the same name that receives the table name as a string and concatenate it in the error message.

### CODE REVIEW

**Comment by carlosfrodrigues** [comment]
Someone can help me? It's failing only in two tests: 
drop_test.go:27
reindex_test.go:53

File: database/errors.go [hunk 1]
```diff
@@ -2,12 +2,15 @@ package database
 
 import (
 	"errors"
+	"fmt"
 )
 
-var (
-	// ErrTableNotFound is returned when the targeted table doesn't exist.
-	ErrTableNotFound = errors.New("table not found")
```
Comment (tie) [inline database/errors.go:9]: I’d suggest to keep the error variable and call `fmt.Errorf("%w: %q", ErrTableNotFound, tableName)` at each fail point.

Then you’d need to replace all `err == ErrTableNotFound ` comparisons with `errors.Is(err, ErrTableNotFound)`.

File: database/transaction_test.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func TestTxTable(t *testing.T) {
 
 		// Getting a table that doesn't exist should fail.
 		_, err = tx.GetTable("unknown")
-		require.EqualError(t, err, database.ErrTableNotFound.Error())
```
Comment (tie) [inline database/transaction_test.go:82]: If we want to check that a particular table does not exist, it’d be better to define the following type

```
type TableNotFoundError struct {
	Name string
}

func (*TableNotFoundError) Is(err error) bool {
	return err == ErrTableNotFound
}
```

Then unwrap the error:

```
var notFoundError TableNotFoundError
if errors.As(err, &notFoundError) {
	fmt.Println(notFoundError.Name)
}
```

File: database/errors.go [hunk 3]
```diff
@@ -2,12 +2,15 @@ package database
...
+// ErrTableNotFound is returned when the targeted table doesn't exist.
+func ErrTableNotFound(tableName string) error {	
+	return fmt.Errorf("%w %s", errors.New("Error: no such table: "), tableName)
```
Comment (tie) [inline database/errors.go:10]: I believe the test fails due to this line—each error has a different memory location, so you can’t compare them directly.

File: sql/query/reindex.go [hunk 4]
```diff
@@ -30,7 +30,8 @@ func (stmt ReIndexStmt) Run(ctx context.Context, tx *database.Transaction, args
 	if err == nil {
 		return res, t.ReIndex()
 	}
-	if err != database.ErrTableNotFound {
+	errReturned  := database.ErrTableNotFound(stmt.TableOrIndexName)
```
Comment (tie) [inline sql/query/reindex.go:33]: In particular, this comparison should have been `errors.Is(err, database.ErrTableNotFound)`.

File: sql/query/drop.go [hunk 5]
```diff
@@ -29,7 +29,8 @@ func (stmt DropTableStmt) Run(ctx context.Context, tx *database.Transaction, arg
 	}
 
 	err := tx.DropTable(stmt.TableName)
-	if err == database.ErrTableNotFound && stmt.IfExists {
+	errReturned := database.ErrTableNotFound(stmt.TableName)
+	if err == errReturned && stmt.IfExists {
```
Comment (tie) [inline sql/query/drop.go:33]: Use `errors.Is` here too.

File: database/errors.go (hunk 1 above)
Comment (asdine) [reply database/errors.go:9]: Very good idea!

**Comment by tie** [comment]
@carlosfrodrigues thanks for the PR! I believe the tests fail because `ErrTableNotFound` function in your change returns a new error each time (both `fmt.Errorf` and `errors.New` return a new pointer). I’ve left a few suggestion in the review.

I also recommend reading [Working with Errors in Go 1.13](https://blog.golang.org/go1.13-errors) blog post.

File: database/errors.go (hunk 1 above)
Comment (carlosfrodrigues) [reply database/errors.go:9]: Thanks for the answer, I'm going to follow your suggestions :)

**Pull request closed without merging**